### PR TITLE
Add JustMoney

### PR DESCRIPTION
**Pull request by salparadise93**
##### Twitter Link:

https://twitter.com/JustMoneyIO

##### Website Link:

https://about.just.money/

##### Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders):

https://about.just.money/assets/img/logos/justmoney.svg

##### Current TVL:

697,317 USD

##### Chain:

Tron, BSC, Polygon, BitTorrent, Zenith

##### Coingecko ID (so your TVL can appear on Coingecko): (https://api.coingecko.com/api/v3/coins/list)

justmoney-2

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)

19946

##### Short Description (to be shown on DefiLlama):

An ecosystem for the next generation of decentralized applications to power Web3 and DeFi.

##### Token address and ticker if any:

BSC, Polygon, BTTC: 0x388D819724dD6d71760A38F00dc01D310d879771
Tron: TVHH59uHVpHzLDMFFpUgCx2dNAQqCzPhcR

Ticker: JM

##### Category (full list at https://defillama.com/categories) *Please choose only one:

Dexes

##### methodology (what is being counted as tvl, how is tvl being calculated):

TVL is calculated based on tokens in pools.

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/justmoney exports TVL: 
        
 
 
    tron                      483.73 k
    bsc                       135.59 k
    bittorrent                2.24 k
    polygon                   43.85 k
    
    total                    665.41 k

**Comment by g1nt0ki** [comment]
@salparadise93 can you change to code to compute tvl using on chain calls instead? if you give me the factory contract addresses, I can help you with it :)

**Comment by salparadise93** [comment]
@g1nt0ki Thanks for the update. I've changed the code to use on chain calls for all chains except Tron. Could you provide any suggestions how it can be easily solved without API calls? Here is the factory contract address: `TBfTeNjh7k8PbkTad8z6WS2vqh7SQZUfQ8`. Also Bittorrent chain returns zeros, probably because of unknown assets.

p.s. I am not the part of the JustMoney team, but I can reach them via Telegram if needed.

**Comment by llamatester** [comment]
The adapter at projects/justmoney exports TVL: 
        
 
 
    bittorrent                0
    bsc                       131.72 k
    polygon                   42.31 k
    tron                      360.68 k
    
    total                    534.71 k

**Comment by llamatester** [comment]
The adapter at projects/justmoney exports TVL: 
        
 
 
    bittorrent                5.32 k
    bsc                       132.38 k
    polygon                   42.20 k
    tron                      353.25 k
    
    total                    533.14 k

File: projects/helper/portedTokens.js [hunk 1]
```diff
@@ -1527,6 +1527,14 @@ const bittorrentFixMapping = {
   "0x9b5f27f6ea9bbd753ce3793a07cba3c74644330d": {
     coingeckoId: "tether",
     decimals: 18
+  },
+  "0x23181f21dea5936e24163ffaba4ea3b316b57f3c": {
+    coingeckoId: 'bittorrent',
+    decimals: 18,
+  },
+  "TNUC9Qb1rRpS5CbWLmNMxXBjyFoydXjWFR": {
```
Comment (g1nt0ki) [inline projects/helper/portedTokens.js:1535]: this mapping is incorrect?

Comment (g1nt0ki) [reply projects/helper/portedTokens.js:1535]: ignore the comment, I see that you are compensating for 


    "0xedf53026aea60f8f75fca25f8830b7e2d6200662":
      "tron:TNUC9Qb1rRpS5CbWLmNMxXBjyFoydXjWFR", // TRX

File: projects/justmoney/index.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  USDT: { 'address': 'TR7NHqjeKQxGTCi8q8ZY4pL8otSzgjLj6t', 'id': 'tether' },
+  USTX: { 'address': 'TYX2iy3i3793YgKU5vqKxDnLpiBMSa5EdV', 'id': '_ustx' },
+  VBS: { 'address': 'TJRc6ZTMhHEPrWPtfsVvXW1mxHPLw1arZo', 'id': '_vbs' },
+  WOX: { 'address': 'TYVFMntFj7xLMxp1CvuXwg9LpPw1dPwWhM', 'id': '_wox' },
+  ZLF: { 'address': 'TXoPCbHtWTerfiNjFBpJdMQqQJoXoT87pq', 'id': '_zlf' },
+}
+
+const pairs = [
+  [ tokens.TRX, tokens.OLDJM, 'TFijK2dCUsrXoZM7PYhFqnM9F4Tac8uBJt' ],
+  [ tokens.TRX, tokens.OLDJM, 'TFijK2dCUsrXoZM7PYhFqnM9F4Tac8uBJt' ],
```
Comment (g1nt0ki) [inline projects/justmoney/index.js:44]: this line is duplicate

**Review by g1nt0ki — CHANGES_REQUESTED** [review]

**Comment by g1nt0ki** [comment]
@salparadise93 really nice work. Bittorrent chain often fails on server, might end up disabling it.

**Comment by g1nt0ki** [comment]
regarding no. of calls, it is not an issue, we batch it and multicall where we can. For tron, I think it is unavoidable.

**Pull request merged**